### PR TITLE
Add parks downloading and drawing

### PR DESCRIPTION
**Pull request by na-ji**
## Description
I added the ability to download every parks from Overpass API in a defined area. After the parks list is downloaded, the frontend will be able to render every parks in bound. 
The parks list is saved in `static/dist/data/parks.json`.
The list is downloaded only once on the first startup after enabling the option. If you need to re-download the list, you just have to delete the file. 

## Motivation and Context

Since we already have S2 cells rendering, I figured one of the last tool missing to help in creating new pokestops/gyms was the drawing of parks areas. 

## Screenshots (if appropriate):
Parks are rendered in green. This screenshot shows an EX raid gym I created a year ago. 
image

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by na-ji** [comment]
Maybe specify that those are EX eligible parks and not every parks :thinking:

**Comment by JimBim89** [comment]
Also nesting parks are important, not sure if these are separate.

**Comment by acocalypso** [comment]
Could be usefull and extend it display nests?

**Comment by na-ji** [comment]
Niantic use different datasets for the nesting parks. So it needs a different query.
I will add it

File: config/config.ini.example [hunk 1]
```diff
@@ -111,6 +111,14 @@
 #madmin-url:                    # MADmin URL e.g. http://127.0.0.1:5001/map (default=None)
 
 
+# Parks
+#######
+# This setting will download all parks from Overpass API and save them to `static/dist/data/parks.json`
+parks                           # enable parks downloading and drawing
```
Comment (daangroot) [inline config/config.ini.example]: Pls comment these out by default.

File: static/js/map.js [hunk 2]
```diff
@@ -3561,7 +3590,8 @@ function loadRawData() {
             'oNeLng': oNeLng,
             'reids': String(isShowAllZoom() ? excludedPokemon : reincludedPokemon),
             'eids': String(getExcludedPokemon()),
-            'prionotify': prionotifyactiv
+            'prionotify': prionotifyactiv,
+            'parks': loadParks
```
Comment (daangroot) [inline static/js/map.js]: I think you forgot to remove this. :)

File: static/js/map.js [hunk 3]
```diff
@@ -3638,6 +3668,74 @@ function updateMap() {
...
+        const inBoundParks = mapData.exEligibleParks.filter(parkPoints => {
+            return parkPoints.some(point => {
+                return map.getBounds().contains(point)
+            })
+        })
+
+        exEligibleParksLayerGroup.clearLayers()
+
+        inBoundParks.forEach(function (park) {
+            L.polygon(park, {'color': 'limegreen'}).addTo(exEligibleParksLayerGroup)
```
Comment (daangroot) [inline static/js/map.js]: Please add the `interactive` option. This disables mouse events, we don't want those for parks. https://leafletjs.com/reference-1.5.0.html#interactive-layer

File: static/js/map.js [hunk 4]
```diff
@@ -3638,6 +3668,74 @@ function updateMap() {
...
+        const inBoundParks = mapData.nestsParks.filter(parkPoints => {
+            return parkPoints.some(point => {
+                return map.getBounds().contains(point)
+            })
+        })
+
+        nestsParksLayerGroup.clearLayers()
+
+        inBoundParks.forEach(function (park) {
+            L.polygon(park, {'color': 'maroon'}).addTo(nestsParksLayerGroup)
```
Comment (daangroot) [inline static/js/map.js]: `interactive` here aswell.

File: templates/map.html [hunk 5]
```diff
@@ -575,6 +575,29 @@ <h3>PokéStop Cells (S2 L17)</h3>
                   </div>
                 </div>
               </div>
+
```
Comment (daangroot) [inline templates/map.html]: Can you remove this white line?

File: templates/map.html [hunk 6]
```diff
@@ -575,6 +575,29 @@ <h3>PokéStop Cells (S2 L17)</h3>
...
+              {% if show.parks %}
```
Comment (daangroot) [inline templates/map.html]: 2 spaces indentation is missing here.

File: templates/map.html [hunk 7]
```diff
@@ -575,6 +575,29 @@ <h3>PokéStop Cells (S2 L17)</h3>
...
+                <h3>Nests Parks</h3>
+                <div class="onoffswitch">
+                  <input id="nests-parks-switch" type="checkbox" name="nests-parks-switch" class="onoffswitch-checkbox">
+                  <label class="onoffswitch-label" for="nests-parks-switch">
+                    <span class="switch-label" data-on="On" data-off="Off"></span>
+                    <span class="switch-handle"></span>
+                  </label>
+                </div>
+              </div>
+              {% endif %}
```
Comment (daangroot) [inline templates/map.html]: Indentation missing here aswell.

**Pull request merged**